src/test/java/org/mockitousage/verification/DelayedExecution.java

```diff
@@ -0,0 +1,51 @@
+package org.mockitousage.verification;
+
+import org.mockitousage.IMethods;
+
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ScheduledExecutorService;
+
+class DelayedExecution {
+    private final ScheduledExecutorService executor;
+    private final IMethods mock;
+    private final int delay;
+    private final ArrayList<Character> invocations = new ArrayList<Character>();
+
+    public DelayedExecution(ScheduledExecutorService executor, IMethods mock, int delay) {
+        this.executor = executor;
+        this.mock = mock;
+        this.delay = delay;
+    }
+
+    public void recordAsyncCall(char c) {
+        invocations.add(c);
+    }
+
+    public void allAsyncCallsStarted() throws InterruptedException {
+        final CountDownLatch countDownLatch = new CountDownLatch(invocations.size());
+        for (final Character invocation : invocations) {
+            executor.execute(runnable(countDownLatch, invocation));
+        }
+        countDownLatch.await();
+    }
+
+    private Runnable runnable(final CountDownLatch countDownLatch, final Character invocation) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                countDownLatch.countDown();
```

Comment:
Yes and no. :)

If I move it as a last line in run method then I guarantee that invocation on mock has been performed and in that case any time verification (later in test) is pretty pointless. It would not make a difference if the time slot is 1 millisecond or 1000 milliseconds.

On the other hand, If I move the countDown before execution on mock (like it is proposed right now), then verification after some time makes sense because we are executing at least two activities in parallel.